edb/schema/pointers.py

```diff
@@ -1539,12 +1543,30 @@ def compile_expr_field(
             else:
                 in_ddl_context_name = None
 
+            # If we are in a link property's default field
+            # do not set path prefix anchor, because link properties
+            # cannot have defaults that reference the object being inserted
+            should_set_path_prefix_anchor = True
+            if field.name == 'default':
+                from .objtypes import ObjectTypeCommandContext
```

Comment:
This is a pretty gnarly way to check if it is a link property.
I would try something like
```
parent_ctx = self.get_referrer_context_or_die(context)
source = parent_ctx.op.get_object(schema, context)
```
and then check if source is a `Pointer`. If it is, this is a link property.
(There are similar checks in `_compile_expr`)